api/src/main/java/com/brandwatch/minibcr/api/controllers/QueryController.java

```diff
@@ -0,0 +1,41 @@
+package com.brandwatch.minibcr.api.controllers;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.brandwatch.minibcr.api.services.QueryService;
+import com.brandwatch.minibcr.common.domain.Query;
+
+@RestController
+public class QueryController {
+
+    @Autowired
+    private QueryService queryService;
+
+    @GetMapping("/queries")
+    public List<Query> getQueries() {
+        return queryService.getQueries();
+    }
+
+    @GetMapping("/queries/{id}")
+    public Query getQueryById(@PathVariable long id) {
```

Comment:
What happens if I pass something that isn't an integer to this endpoint?
Or a queryID that doesn't exist?

Reply:
At the moment it would be caught by the query repository and it will log an error message to the console like: 
`2021-08-18 12:09:00.571  INFO 77021 --- [nio-8080-exec-1] c.b.m.c.r.MentionRepositoryImpl          : Mention with id 2 does not exist...`
I'll add a better client response scheme once I start crawling real data as I dont know for now what parameters that data will contain

Reply:
if something other than an integer is passed, it treats it as an incorrect URL. I am not sure if that's the expected behaviour

Reply:
It might be good to validate manually, and throw an error to the user, explaining what the parameter should be. 

I'd create another method, like `validateInput(long id)` or something named more appropriately. You could then throw an [IllegalArgumentException](https://docs.oracle.com/javase/7/docs/api/java/lang/IllegalArgumentException.html).